game-core/src/main/java/games/strategy/triplea/ai/pro/util/ProOddsCalculator.java

```diff
@@ -217,23 +228,40 @@ public ProBattleResult callBattleCalculator(
     double tuvSwing =
         results.getAverageTuvSwing(
             attacker, mainCombatAttackers, defender, mainCombatDefenders, data);
-    if (Matches.territoryIsNeutralButNotWater().test(t)) { // Set TUV swing for neutrals
+
+    // Set TUV swing for neutrals
+    if (Matches.territoryIsNeutralButNotWater().test(t)) {
       final double attackingUnitValue = TuvUtils.getTuv(mainCombatAttackers, ProData.unitValueMap);
       final double remainingUnitValue =
           results
               .getAverageTuvOfUnitsLeftOver(ProData.unitValueMap, ProData.unitValueMap)
               .getFirst();
       tuvSwing = remainingUnitValue - attackingUnitValue;
     }
+
+    // Add TUV swing for transported units
     final List<Unit> defendingTransportedUnits =
         CollectionUtils.getMatches(defendingUnits, Matches.unitIsBeingTransported());
-    if (t.isWater()
-        && !defendingTransportedUnits.isEmpty()) { // Add TUV swing for transported units
+    if (t.isWater() && !defendingTransportedUnits.isEmpty()) {
       final double transportedUnitValue =
           TuvUtils.getTuv(defendingTransportedUnits, ProData.unitValueMap);
       tuvSwing += transportedUnitValue * winPercentage / 100;
     }
 
+    // Remove TUV and add to remaining units for defenders that can submerge before battle
+    if (tuvSwing > 0
+        && checkSubmerge
+        && Properties.getSubRetreatBeforeBattle(data)
+        && defendingUnits.stream().anyMatch(Matches.unitCanEvade())
+        && attackingUnits.stream().noneMatch(Matches.unitIsDestroyer())) {
```

Comment:
side-note, it really seems like there should a rules class to do this logic. If there are other units that can prevent subs from submerging, seems this would break. 

Does v2 or v1 rules allow destroyers to block sub-submerge? It might be moot if there are no destroyers on those maps, but could be worth considering that the 'isDestroyer' condition is truly universal.

Reply:
The isDestroyer is actually checking the unit property not whether the unit itself is a destroyer. I can't remember if v1 handles it the same as the rest but at the moment all the AI logic assumes that isDestroyer property prevents submerging subs.

Reply:
I think v1 can get away with it because there are no destroyer units.